### PR TITLE
Add ignoreMediaFeatureNames for unit-blacklist

### PR DESCRIPTION
**Pull request by lucmerceron**
> Which issue, if any, is this issue related to?

#3006 

> Is there anything in the PR that needs further explanation?

The PR is easy to understand.

NB:
I am not sure this solution is the most optimized since I have one loop for the value `valueParser` inside the one for the media `mediaParser` but it feels safer than doing matches only on the returned `mediaFeatureNode` and having to add `column` size for error handling.

Since this is my first PR for stylelint, this is widely open to discussion. I may have missed a util or anything else that could be used.

### CODE REVIEW

File: lib/rules/unit-blacklist/index.js [hunk 1]
```diff
@@ -31,18 +40,64 @@ const rule = function(blacklistInput, options) {
...
         }
       }
     );
     if (!validOptions) {
       return;
     }
 
-    function check(node, value, getIndex) {
+    function checkMedia(node, value, getIndex) {
+      value = value.replace(/\*/g, ",");
```
Comment (CAYdenberg) [inline lib/rules/unit-blacklist/index.js]: Let's leave this line out. Its sole purpose is handle parsing of calc functions - I don't think it's needed within media queries (and it's a bit of a hack so I'd like to not proliferate it if possible).

File: lib/rules/unit-blacklist/index.js [hunk 2]
```diff
@@ -31,18 +40,64 @@ const rule = function(blacklistInput, options) {
...
+          // Ignore all non-word valueNode and
+          // the values not included in the parentValue string
+          if (
+            valueNode.type !== "word" ||
+            !parentValue.includes(valueNode.value)
+          ) {
+            return;
+          }
+
+          const unit = getUnitFromValueNode(valueNode);
```
Comment (CAYdenberg) [inline lib/rules/unit-blacklist/index.js]: From here to the end of the function is almost a mirror image of the same thing in walk.decls. I think it should be abstracted.

**Comment by CAYdenberg** [comment]
Thanks. Tests and documentation looks good.

File: lib/rules/unit-blacklist/index.js (hunk 1 above)
Comment (lucmerceron) [reply lib/rules/unit-blacklist/index.js]: Done in my last commit b6effdf

File: lib/rules/unit-blacklist/index.js (hunk 2 above)
Comment (lucmerceron) [reply lib/rules/unit-blacklist/index.js]: Done in my last commit too b6effdf

**Comment by CAYdenberg** [comment]
Awesome. Thanks again.

File: lib/rules/unit-blacklist/README.md [hunk 3]
```diff
@@ -94,3 +94,46 @@ a { -moz-border-radius-topright: 40px; }
...
+```
+
+```css
+@media not (resolution: 300dpi) { }
+```
+
+The following patterns are considered violations:
+
+```css
+a { @media screen and (max-device-width: 500px) { } }
```
Comment (jeddy3) [inline lib/rules/unit-blacklist/README.md]: Can we just have `@media screen and (max-device-width: 500px) {}` here please as I don't believe nested media queries are standard syntax.

File: lib/rules/unit-blacklist/README.md [hunk 4]
```diff
@@ -94,3 +94,46 @@ a { -moz-border-radius-topright: 40px; }
...
+{
+  "px": [ "min-width", "/height$/" ],
+  "dpi": [ "resolution" ]  
+}
+```
+
+The following patterns are *not* considered violations:
+
+```css
+@media (min-width: 960px) { }
```
Comment (jeddy3) [inline lib/rules/unit-blacklist/README.md]: Can you please change the examples to use empty braces i.e. `{}` as per [the guidelines](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#example-patterns).

Comment (lucmerceron) [reply lib/rules/unit-blacklist/README.md]: Yep, sorry about that. Done in my last [commit](6a0bc46).

File: lib/rules/unit-blacklist/README.md (hunk 3 above)
Comment (lucmerceron) [reply lib/rules/unit-blacklist/README.md]: Done in my last [commit](6a0bc46)

**Pull request merged**